### DISCUSSION TITLE
fatal - AssertionError [ERR_ASSERTION]: Invalid config values: fastRefresh

### DISCUSSION THREAD

**Discussion by jiangxiaoqiang** [umijs/umi#8391] [Q&A]
有朋友遇到过这个问题么：https://stackoverflow.com/questions/72865167/fatal-assertionerror-err-assertion-invalid-config-values-fastrefresh 尝试过这个方法：https://github.com/ant-design/ant-design/issues/33957 但是没有修复,调整为true的时候，提示错误：

```
$ umi g tmp
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
Validate config "fastRefresh" failed, "value" must be of type object
ValidationError: "value" must be of type object
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Error: Process completed with exit code 1.

```

**Comment by fz6m** [comment]
你需要关闭 fast refresh 吗？这个配置项必须是 boolean 值。

**Reply by jiangxiaoqiang** [reply]
实际的配置是这样的：`fastRefresh: {},`，https://v3.umijs.org/zh-CN/docs/fast-refresh

**Reply by fz6m** [reply]
你用的是 umi3 还是 4 ？你提到的这个 stackoverflow 问题是 umi4 ，必须是 boolean，umi3 需要是 object 。

**Reply by 664542228** [reply]
启动命令
如果使用了 @umijs/max 可以使用 max 命令来替换 umi，max dev，max build 等

umi@4 将一些项目前置操作放到了 setup 命令中，如 umi@3 中的 umi g tmp 等命令，需要使用 umi setup 替换

package.json

{
  "scripts": {
-    "build": "umi build",
+    "build": "max build",
-    "postinstall": "umi g tmp",
+    "postinstall": "max setup",
-    "start": "umi dev",
+    "start": "max dev",
  }
}

**Reply by xu-Aurora** [reply]
> 你用的是 umi3 还是 4 ？你提到的这个 stackoverflow 问题是 umi4 ，必须是 boolean，umi3 需要是 object 。

我是umi3也碰到这个问题了，请问是什么原因？

**Comment by fz6m** [comment]
umi 3 已经不维护了，推荐升级到 umi 4
